Homework 3/script.js

```diff
@@ -39,12 +51,56 @@ $( document ).ready(function() {
 	});
 
 	$("#signUpButton").click(function() {
+		$("#signUpErrorMessage").hide();
+		$("#signUpErrorMessage").empty();
+		if(($('#signUpPasswordInput').val() != '') && ($('#signUpPasswordConfirmInput').val() != '') && ($('#signUpEmailInput').val() != '') && ($('#signUpUsernameInput').val() != '')){
+			if($('#signUpPasswordInput').val() == $('#signUpPasswordConfirmInput').val()){
+				if($('input[name=accountType]:checked').val() == "storeOwner"){
+					if(($('#signUpStoreNameInput').val() != '') && ($('#signUpStoreImageUrlInput').val() != '')){
+						productManager.createUser($('#signUpUsernameInput').val(), $('#signUpEmailInput').val(), $('#signUpPasswordInput').val(), true, function(success, createdUser){
+							if(success){
+								productManager.createStore($('#signUpStoreNameInput').val(), $('#signUpStoreImageUrlInput').val(), ($('input[name=isOpen]:checked').val() == "true"), function(store){
+									user = new StoreOwner(createdUser, store);
+									productManager.createStoreUserRelation(store.cloudboostObject, createdUser, function(){
+										loadMallPage();
+									})
+								});
+							}
+						});
+					}
+					else{
+						$("#signUpErrorMessage").show();
+						$("#signUpErrorMessage").append("* You must fill in all of the fields.");
+					}
+				}
+				else{
+					productManager.createUser($('#signUpUsernameInput').val(), $('#signUpEmailInput').val(), $('#signUpPasswordInput').val(), false, function(success, createdUser){
+						if(success){
+							user = createdUser;
+							loadMallPage();
+						}
+					});
+				}
+			}
+			else{
+				$("#signUpErrorMessage").show();
+				$("#signUpErrorMessage").append("* Password and password confirm do not match.");
+			}
+		}
+		else{
+			$("#signUpErrorMessage").show();
+			$("#signUpErrorMessage").append("* You must fill in all of the fields.");
+		}
 	});
 
 	$("#signOutButton").click(function() {
 		location.reload();
 	});
 
+	$("#ownerSignOutButton").click(function() {
```

Comment:
Again css selectors can be chained. Study up CSS selectors a bit. http://www.w3schools.com/cssref/css_selectors.asp

```js
$("#signOutButton, #ownerSignOutButton")
```